Src/AutoMoq/MockType.cs

```diff
@@ -68,11 +68,14 @@ internal static Type GetMockedType(this Type type)
             where TMock : class
         {
             return setup.Returns(() =>
-                {
-                    var result = (TResult) context.Resolve(typeof (TResult));
-                    setup.Returns(result);
-                    return result;
-                });
+            {
+                var specimen = context.Resolve(typeof (TResult));
+                TResult result = specimen is TResult ? (TResult) specimen : default(TResult);

```

Comment:
What happens if `specimen` is `OmitSpecimen` and `TResult` is a reference type? Wouldn't that cause `result` to be `null`?